htlcswitch/hop/payload.go

```diff
@@ -250,6 +254,12 @@ func (h *Payload) MultiPath() *record.MPP {
 	return h.MPP
 }
 
+// AMPRecord returns the record corresponding with option_amp parsed from the
+// onion payload.
+func (h *Payload) AMPRecord() *record.AMP {
+	return h.AMP
+}
```

Comment:
I’ve intentionally left those commits out of this PR so that this logic can’t be hit from the network, and can be feature flipped once more of the surrounding logic has been merged. It’s a ending AMP data in at this point gains is much, we have mock payloads for that, and it is more likely to leave the db in a weird state because there is more logic waiting to be merged after this PR.

Reply:
It is very much intentional that you can’t hit this logic externally by any means other than legacy keysend.

Reply:
Ok, makes sense